custom_components/airthings_wave/airthings.py

```diff
@@ -192,152 +206,158 @@ def __init__(self, scan_interval, mac=None):
         self.scan_interval = scan_interval
         self.last_scan = -1
         self._dev = None
-
-    def _parse_serial_number(self, manufacturer_data):
-        try:
-            (ID, SN, _) = struct.unpack("<HLH", manufacturer_data)
-        except Exception as e:  # Return None for non-Airthings devices
-            return None
-        else:  # Executes only if try-block succeeds
-            if ID == 0x0334:
-                return SN
-
-    def find_devices(self, scans=50, timeout=0.1):
+        self._command_data = None
+
+    def notification_handler(self, sender, data):
+        _LOGGER.debug("Notification handler: {0}: {1}".format(sender, data))
+        self._command_data = data
+        self._event.set()
+    
+    async def find_devices(self, scans=2, timeout=5):
         # Search for devices, scan for BLE devices scans times for timeout seconds
         # Get manufacturer data and try to match it to airthings ID.
-        scanner = btle.Scanner()
+        
+        _LOGGER.debug("Scanning for airthings devices")
         for _count in range(scans):
-            advertisements = scanner.scan(timeout)
+            advertisements = await BleakScanner.discover(timeout)
             for adv in advertisements:
-                sn = self._parse_serial_number(adv.getValue(btle.ScanEntry.MANUFACTURER))
-                if sn is not None:
-                    if adv.addr not in self.airthing_devices:
-                        self.airthing_devices.append(adv.addr)
+                if 820 in adv.metadata["manufacturer_data"]: # TODO: Not sure if this is the best way to identify Airthings devices
+                    if adv.address not in self.airthing_devices:
+                        self.airthing_devices.append(adv.address)
 
         _LOGGER.debug("Found {} airthings devices".format(len(self.airthing_devices)))
         return len(self.airthing_devices)
 
-    def connect(self, mac, retries=10):  
+    async def connect(self, mac, retries=10):  
+        _LOGGER.debug("Connecting to {}".format(mac))
+        await self.disconnect()
         tries = 0
-        self.disconnect()
         while (tries < retries):
             tries += 1
             try:
-                self._dev = btle.Peripheral(mac.lower())
-                self.delgate = MyDelegate()
-                self._dev.withDelegate( self.delgate )
+                self._dev = BleakClient(mac.lower())
+                ret = await self._dev.connect()
+                if ret:
+                    _LOGGER.debug("Connected to {}".format(mac))
```

Comment:
Retry if not connected?

Reply:
good finding. @mjmccans maybe you can also pull this into your repo